nautobot/extras/views.py

```diff
@@ -455,6 +457,25 @@ class CustomFieldBulkDeleteView(generic.BulkDeleteView):
     queryset = CustomField.objects.all()
     table = tables.CustomFieldTable
 
+    def construct_custom_field_delete_tasks(self, queryset):
+        """
+        Helper method to construct a list of celery tasks to execute when bulk deleting custom fields.
+        """
+        tasks = [
+            delete_custom_field_data.si(obj.name, set(obj.content_types.values_list("pk", flat=True)))
+            for obj in queryset
+        ]
+        return tasks
+
+    def perform_pre_delete(self, request, queryset):
+        if not get_worker_count(request):
+            messages.error(request, "Unable to run job: Celery worker process not running.")
+            return redirect(self.get_return_url(request))
```

Comment:
This doesn't actually do anything, right? Because `BulkDeleteView.post()` doesn't actually check the return value from `perform_pre_delete()`.

Reply:
I tried it with `celery_worker_1` stopped, it didnt actually delete the `_custom_field_data` from the device and went back to the return_url. so I think this if statement is def needed.

Reply:
That's because the base `BulkDeleteView.post()` also redirects to that same URL. This `return redirect(...)` isn't doing anything with the redirect as far as I can see, it should be just `return`.

Reply:
I see understood. Should I get rid of the `mock` decorators too then since the function is going to return None regardless. Or do we want to keep them?

Reply:
Removed the unittest overrides with `mock` decorators in the latest commit.